guides/source/5_1_release_notes.md

```diff
@@ -0,0 +1,98 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON http://guides.rubyonrails.org.**
+
+Ruby on Rails 5.1 Release Notes
+===============================
+
+Highlights in Rails 5.1:
+
+* Yarn Support
+* Optional Webpack support
+* jQuery no longer a default dependency
+* System tests
+* Encrypted secrets
+* Parameterized mailers
+* Direct & resolved routes
+* Unification of form_for and form_tag into form_with
+
+These release notes cover only the major changes. To learn about various bug
+fixes and changes, please refer to the change logs or check out the [list of
+commits](https://github.com/rails/rails/commits/5-1-stable) in the main Rails
+repository on GitHub.
+
+--------------------------------------------------------------------------------
+
+Upgrading to Rails 5.1
```

Comment:
We need to add info about the HashWithIndifferentAccess deprecation. This is already [mentioned on the upgrading page](https://github.com/rails/rails/blob/master/guides/source/upgrading_ruby_on_rails.md)

I believe it should be mentioned and guided here as-well, although it's only a deprecation note for future 5.2-release.